test/e2e/epp/e2e_test.go

```diff
@@ -31,20 +35,91 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
+	v1 "sigs.k8s.io/gateway-api-inference-extension/api/v1"
 	"sigs.k8s.io/gateway-api-inference-extension/apix/v1alpha2"
 	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/metadata"
 	testutils "sigs.k8s.io/gateway-api-inference-extension/test/utils"
 )
 
+const (
+	firstPort             = 8000
+	numPorts              = 2
+	maxConcurrentRequests = 2 // prevent hammering Envoy and backend
+	maxRetries            = 3
+	backoff               = 1 * time.Second
+	safetyMult            = 3 // increase number of batches to account for retries
+)
+
 var _ = ginkgo.Describe("InferencePool", func() {
 	var infObjective *v1alpha2.InferenceObjective
 	ginkgo.BeforeEach(func() {
 		ginkgo.By("Waiting for the namespace to exist.")
 		namespaceExists(testConfig)
 
+		ginkgo.By("Modifying deployment using local image for testing (temporary).")
+		deploy := &appsv1.Deployment{}
+		key := types.NamespacedName{Name: modelServerName, Namespace: testConfig.NsName}
+
+		gomega.Eventually(func() error {
+			err := testConfig.K8sClient.Get(testConfig.Context, key, deploy)
+			if err != nil {
+				return err
+			}
+
+			// Instead of hardcoding arguments, we can instead replace the arguments that need
+			// to be changed, preserving any others that may exist.
+			var newArgs []string
+			skipNext := false
```

Comment:
If you move to a dedicated manifest file, this entire block of code disappears, making the test much cleaner and easier to maintain.